### ISSUE TITLE
Mafia - City of Lost Heaven (TLB misses)

### ISSUE THREAD

**Issue by Blackbird88** [PCSX2/pcsx2#1496]
-PCSX2 version: 1.5.0 GIT

-PCSX2 options: Defaults + the game relies on **EE Timing Hack** otherwise it hangs on Start menu.

-Plugins used: GSdx, SPU2-X, LilyPad

-Plugin settings: Defaults, no settings modifications affect the issue

-Description: TLB misses after finishing **The Running Man** mission.
![image](https://cloud.githubusercontent.com/assets/4414625/17323768/9ba09e32-58a3-11e6-830d-af06fbf751f3.png)
[Full log](https://gist.github.com/Blackbird88/f0261b8602c81603b5b02166eb9d732f)

-How to replicate: Finish the **The Running Man** mission and let **Molotov Party** to load.

-Last known version to work: Never worked

**Memorycard save**
http://www.mediafire.com/download/g532hvi59vptbkw/Mcd002.zip
**Save state**
http://www.mediafire.com/download/mp9azhi2t1n5hrm/SLUS-20671_%28B67F4F9E%29.00.zip

**Comment by FlatOutPS2** [comment]
Did you play the game from the original DVD?

**Comment by Blackbird88** [comment]
Yes. There is no issue with the dump or the DVD.

**Comment by FlatOutPS2** [comment]
And just to be safe, did you try playing without widescreen patches enabled too?

**Comment by Blackbird88** [comment]
Yes. Also no. Fiddling with settings is waste of time. Been there, done that.

**Comment by FlatOutPS2** [comment]
I'm sure you tried everything, but widescreen patches (and cheats) are the cause of most TLB miss issues, and as they're turned on in the log you posted I had to ask.

I added the memorycard save from the compatibility report to the first post, so it's easier for others to test and fix it.

**Comment by Blackbird88** [comment]
I will test them one more time then! But it's unlikely.
Also good idea. I hadn't though of that. Thanks!

**Comment by prafullpcsx2** [comment]
Can you post a memory card save as well as a savestate before the crash point? I will have a look at this and try to find a solution.

**Comment by Blackbird88** [comment]
Sure. Added to first post. Just skip the FMVs and let the Molotov Party load. Thanks for taking interest!

**Comment by prafullpcsx2** [comment]
@Blackbird88 Can you try this patch? I am little short on time tonite so can't check fully myself.

http://forums.pcsx2.net/attachment.php?aid=60672

The patch should work but it may result in some camera problems (other problems too). Can you please play the game a bit and may be even finish this level with the patch enabled?

**Comment by Blackbird88** [comment]
Worked without problems to me (as far as I could see). The crash is gone. Amazing work!

**Comment by ramapcsx2** [comment]
patch=1,EE,003e2bb4,word,00000000

What are you nop'ing there and why? :)

**Comment by Blackbird88** [comment]
Okay so I was trying to run the game without the patch ans it seems to happen in certain missions too meaning this patch fixes more than just **Molotov Party**

**Comment by prafullpcsx2** [comment]
@Blackbird88 : Savestate before the crash please :)

**Comment by Blackbird88** [comment]
Here
http://www.mediafire.com/download/ad3dw1c8he2wsdd/SLUS-20671_%28B67F4F9E%29.00_THEPRIEST.zip

**Comment by prafullpcsx2** [comment]
@Blackbird88 : Can you try this patch and report back?

http://forums.pcsx2.net/attachment.php?aid=61010

**Comment by Blackbird88** [comment]
Yup, works. Just finished the mission. Thanks!

**Comment by prafullpcsx2** [comment]
Good to hear that. Let me know if you run into further problems with the game. Also check whether the patch creates problems in some other part of game. If you manage to finish the game, do update the compatibility report on the forums.

**Comment by Blackbird88** [comment]
Juts finished the main storyline with no crashes. Need to test Racing mode too just in case (doubt it's broken tho since Freeride is also fine)
![image](https://cloud.githubusercontent.com/assets/4414625/19269902/532ff5d4-8fbd-11e6-9414-d698fbd83ca2.png)

**Comment by gregory38** [comment]
Is the issue still valid/open ?

**Comment by Blackbird88** [comment]
Well I'm not sure if the game is playable from start to end with both of those patches since I only tested them on the fly.

**Comment by MrCK1** [comment]
@Blackbird88 When you have some free time, can you do a playthrough with Prafull's patch again just to confirm the status? I also added a link to the patch on the game's wiki page.

**Comment by Blackbird88** [comment]
I'll do it I promise. It's just that I didn't get around it yet due to IRL stuff :/

**Comment by Davo6233** [comment]
Hello Blackbird88, I have same problem but these solutions dont work for me. Maybe because I'm new for PCSX2. Can you please upload somewhere whole folder with PCSX2 and Your mafia game and share it with me ?? Because I cant solve the problem. Or explain the solution more detailed.
Thank you of reply ! :-)

**Comment by MrCK1** [comment]
@Davo6233 Please redirect your questions to forums.pcsx2.net. This is not a place for for general support. Thanks!

**Comment by Krusher97** [comment]
I have a US copy of the game myself. To test this do I need to apply both the patches and complete the game from the beginning to end?

**Comment by Blackbird88** [comment]
Yes. There are two known crashes.

**Comment by MrCK1** [comment]
@Krusher97 The most recent patch should do: http://forums.pcsx2.net/attachment.php?aid=61010

Ideally from beginning to end would be best, since the game crashes randomly. Also, It would be a great help if you could update the compatibility report in the forums when you're finished ;)

**Comment by Krusher97** [comment]
Does it matter if I update the thread or the wiki?

**Comment by MrCK1** [comment]
The thread would be nice better :)

**Comment by Davo6233** [comment]
SOLVED!! Thx guys, i totaly forgot that I need an USA version of the game. (Cuz I had EU version) So now it works!, thank you again.

**Comment by Krusher97** [comment]
Finished the game from beginning to end with no crashes or freezes with the two cheats enabled. One weird thing of note was how the Software renderer always ran better than the hardware one no matter what Internal resolution.

I will update the compatibility thread when these fixes get merged.

**Comment by MrCK1** [comment]
@Krusher97 Perfect. Thank you! ;)

**Comment by atomic83GitHub** [comment]
Hum, Just for the curiosity, this PR don't solve the issue?
https://github.com/PCSX2/pcsx2/pull/1783

**Comment by Blackbird88** [comment]
It doesn't. Still crashes when loading Molotov Party.

**Comment by atomic83GitHub** [comment]
I am not in favor of closing that issue. Yes there is a patch but it makes loading screens longer compared to a real console and break the loading bar (minor but a bug is a bug).

As a side note, that issue is common accros every single ps2 emulators so far.

**Comment by refractionpcsx2** [comment]
Can somebody check this on the latest master? (1.7.4486 or higher)

I'm curious if it still crashes without the patch. Apparently both just before the Molotov Party and The Priest Mission (whatever that is) are both broken.

Remember to enable advanced settings and untick Compatibility Patches, so it doesn't enable the patch :)

**Comment by Blackbird88** [comment]
> 

Still crashes if the patch is disabled.

**Comment by refractionpcsx2** [comment]
bummer, ah well, was worth a shot.
